### PR TITLE
feat: add http/2 support

### PR DESCRIPTION
**Pull request by kemingy**
`hyper` can serve the traffic with HTTP/1.1 & HTTP/2.

Verified with `curl -I --http2-prior-knowledge -w '%{http_version}\n' http://127.0.0.1:8000/metrics`.

### CODE REVIEW

**Comment by kemingy** [comment]
For Python `httpx`, you will need:

```python
import httpx

# force to use HTTP/2
client = httpx.Client(http1=False, http2=True)
resp = client.post('http://127.0.0.1:8000/inference', json={'time': 0})

print(resp.http_version)
```

- refer to https://github.com/encode/httpx/issues/3261#issuecomment-2288779077

**Comment by lkevinzc** [comment]
Now the server will only support http/2? Will it be better to let user configure?

**Comment by kemingy** [comment]
It supports both. Depends on which protocol the client uses.

**Review by lkevinzc — APPROVED** [review]

File: README.md [hunk 1]
```diff
@@ -229,6 +229,7 @@ More ready-to-use examples can be found in the [Example](https://mosecorg.github
...
 - You should choose appropriate `serialize/deserialize` methods, which are used to decode the user request and encode the response. By default, both are using JSON. However, images and embeddings are not well supported by JSON. You can choose msgpack which is faster and binary compatible (ref [Stable Diffusion](https://mosecorg.github.io/mosec/examples/stable_diffusion.html)).
 - Configure the threads for OpenBLAS or MKL. It might not be able to choose the most suitable CPUs used by the current Python process. You can configure it for each worker by using the [env](https://mosecorg.github.io/mosec/reference/interface.html#mosec.server.Server.append_worker) (ref [custom GPU allocation](https://mosecorg.github.io/mosec/examples/env.html)).
+- Switch to the HTTP/2 client. `mosec` supports HTTP/2 since v0.8.8.
```
Comment (lkevinzc) [inline README.md:232]:
```suggestion
- Enable HTTP/2 from client side. `mosec` automatically adapts to user's protocol (e.g., HTTP/2) since v0.8.8.
```

**Pull request merged**